staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go

```diff
@@ -1101,6 +1101,9 @@ func (ss *scaleSet) EnsureHostsInPool(service *v1.Service, nodes []*v1.Node, bac
 
 	// Update VMs with best effort that have already been added to nodeUpdates.
 	for meta, update := range nodeUpdates {
+		// create new instance of meta and update for passing to anonymous function
```

Comment:
I don't suppose we could UT this to ensure that the `meta` and `update` variable references are discrete in the callback functions inside this enumeration? (or maybe they aren't callbacks per se, but let's make this implemented in such a way that we don't erroneously hold a common reference in the stack when we think the references are distinct)

Reply:
agreed, this is hard to catch and because AggregateGoroutines() is used here, we couldn't pass the values as parameters.